Extension/src/Utility/Process/program.ts

```diff
@@ -0,0 +1,172 @@
+    }
+
+    let result: ProcessFunction | CommandFunction;
+    if (opts.noninteractive) {
+        // create launcher for non-interactive commands
+        result = (async (...moreArgs: ArrayPlusOptions<Primitive, ProgramOptions>): Promise<CommandResult> => {
+            // make sure the path is executable
+
+            // Create Process instance
+            const moreOpts = options(moreArgs);
+            const subscribers = opts.on ? moreOpts.on ? [opts.on, moreOpts.on] : [opts.on] : moreOpts.on ? [moreOpts.on] : [];
+
+            const proc = await new Process(
+                await fullPath,                                                                                   // executable
+                [...result.cmdlineArgs, ...primitives(moreArgs)],                                                  // arguments
+                result.options.cwd || moreOpts.cwd || process.cwd(),                                              // cwd
+                { ...process.env, ...result.options.env, ...moreOpts.env },                                       // environment
+                false,                                                                                            // no stdin.
+                ...subscribers                                                                                    // event handlers
+            );
+
+            const code = await proc.exitCode;
+            return {
+                code,
+                stdio: proc.stdio,
+                error: proc.error
+            };
+
+        }) as any as CommandFunction;
+    } else {
+        // create Launcher function
+        result = (async (...moreArgs: ArrayPlusOptions<Primitive, ProgramOptions>): Promise<Process> => {
+            // Create Process instance
+            const moreOpts = options(moreArgs);
+            const subscribers = opts.on ? moreOpts.on ? [opts.on, moreOpts.on] : [opts.on] : moreOpts.on ? [moreOpts.on] : [];
+            return new Process(
+                await fullPath,                                                                                   // executable
+                [...result.cmdlineArgs, ...primitives(moreArgs)],                                                  // arguments
+                result.options.cwd || moreOpts.cwd || process.cwd(),                                              // cwd
+                { ...process.env, ...result.options.env, ...moreOpts.env },                                       // environment
+                true,
+                ...subscribers                                                                                    // event handlers
+            );
+        }) as any as ProcessFunction;
+    }
+
+    result.cmdlineArgs = cmdlineArgs; // bind the default args to the function
+    result.options = opts;            // bind the default options to the function
+    result.executable = fullPath;         // bind the executable to the function

```

Comment:
Are the comments supposed to be aligned here?

Reply:
With all the refactoring having happened comments may be out of alignment. I'll go thru everything and make sure there are no extra spaces.

Reply:
reducing whitespace to 1